client/ayon_deadline/repository/custom/plugins/GlobalJobPreLoad.py

```diff
@@ -511,113 +515,64 @@ def inject_ayon_environment(deadlinePlugin):
-                if value:
-                    break
-            add_kwargs[key] = value
-
-        if not all(add_kwargs.values()):
-            raise RuntimeError((
-                "Missing required env vars: AYON_PROJECT_NAME,"
-                " AYON_FOLDER_PATH, AYON_TASK_NAME, AYON_APP_NAME"
-            ))
-
-        # Use applications addon arguments
-        # TODO validate if applications addon should be used
-        args = [
-            "--headless",
-            "addon",
-            "applications",
-            "extractenvironments",
-            export_url
-        ]
-
-        # staging requires passing argument
-        # TODO could be removed when PR in ayon-core starts to fill
-        #  'AYON_USE_STAGING' (https://github.com/ynput/ayon-core/pull/1130)
-        #  - add requirement for "core>=1.1.1" to 'package.py' when removed
-        settings_variant = job.GetJobEnvironmentKeyValue(
-            "AYON_DEFAULT_SETTINGS_VARIANT"
-        )
-        if settings_variant == "staging":
-            args.append("--use-staging")
-
-        # Backwards compatibility for older versions
-        legacy_args = [
-            "--headless",
-            "extractenvironments",
-            export_url
-        ]
-
-        for key, value in add_kwargs.items():
-            args.extend(["--{}".format(key), value])
-            # Legacy arguments expect '--asset' instead of '--folder'
-            if key == "folder":
-                key = "asset"
-            legacy_args.extend(["--{}".format(key), value])
-
-        environment = {
-            "AYON_SERVER_URL": ayon_server_url,
-            "AYON_API_KEY": ayon_api_key,
-            "AYON_BUNDLE_NAME": ayon_bundle_name,
-        }
-
-        for key in ("AYON_USE_STAGING", "AYON_IN_TESTS"):
-            value = job.GetJobEnvironmentKeyValue(key)
-            if value:
-                environment[key] = value
+            raise RuntimeError(
+                (
+                    "Missing required values for server url and api key. "
+                    "Please fill in AYON Deadline plugin or provide by "
+                    "AYON_SERVER_URL and AYON_API_KEY"
+                )
+            )
 
-        for env, val in environment.items():
-            # Add the env var for the Render Plugin that is about to render
-            deadlinePlugin.SetEnvironmentVariable(env, val)
-            # Add the env var for current calls to `DeadlinePlugin.RunProcess`
-            deadlinePlugin.SetProcessEnvironmentVariable(env, val)
+        output_dir = _get_output_dir(job)
+
+        shared_env_group = os.environ.get("AYON_SITE_ID")
+        # drive caching of environment variables with env var
+        # it is recommended to use same value AYON_SITE_ID for 'same'
+        # render nodes (eg. same OS etc.)
+        if shared_env_group:
+            print("Caching of environment file will be used.")
+            environment_file_name = f"env_{shared_env_group}.json"
```

Comment:
Shouldn't we use some hash based on project name, folder path, task name and env group so if the same output path is used for different contexts it still can operate?

Reply:
Added `job.JobId` to differentiate.